### PR TITLE
add one shot det demo

### PR DESCRIPTION
**Pull request by MARMOTatZJU**
1.  add one-shot detection [demo](repo_root/demo).
2. resolve all conflicts.

P.S. 1: reviewers are recommended to view from commit __fix docs & shufflenet config 25e15db__ as all commit histories earlier than this point has been squashed into initial commit for training code.
P.S. 2: New commits in dev have been merged into xuyinda-dev (personal develop branch) with all conflicts resolved.

### CODE REVIEW

File: demo/main/debug.py [hunk 1]
```diff
@@ -0,0 +1,4 @@
+from paths import ROOT_PATH  # isort:skip
```
Comment (lzx1413) [inline demo/main/debug.py]: this file should be removed

File: demo/main/osdet/osdet_demo.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
+# -*- coding: utf-8 -*
```
Comment (lzx1413) [inline demo/main/osdet/osdet_demo.py]: please rename this file

File: demo/main/osdet/osdet_demo2.py [hunk 3]
```diff
@@ -0,0 +1,137 @@
+# -*- coding: utf-8 -*
```
Comment (lzx1413) [inline demo/main/osdet/osdet_demo2.py]: please rename this file

File: demo/main/static_image/static_image.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    cv2.putText(im_, "template patch region", (20, 40), cv2.FONT_HERSHEY_COMPLEX, font_size, color["template"], font_width)
+    cv2.putText(im_, "search patch region", (20, 60), cv2.FONT_HERSHEY_COMPLEX, font_size, color["search"], font_width)
+
+    cv2.imshow("im_", im_)
+
+    cv2.imshow("template", im_z)
+    cv2.imshow("search", im_x)
+    cv2.waitKey(0)
+
+    from IPython import embed;embed()
```
Comment (lzx1413) [inline demo/main/static_image/static_image.py]: should be removed

File: demo/resources/static_img_example/getImage.py [hunk 5]
```diff
@@ -0,0 +1,49 @@
+# -*- coding: utf-8 -*
```
Comment (lzx1413) [inline demo/resources/static_img_example/getImage.py]: flile name shoud be get_image.py

File: demo/resources/static_img_example/getImage.py [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    return arr
+
+arr_z = imarray_to_tensor(im_z)
+arr_x = imarray_to_tensor(im_x)
+
+# change color order because deploy model do not support it
+# arr_z = arr_z[:, [2, 1, 0], :, :]
+# arr_x = arr_x[:, [2, 1, 0], :, :]
+
+if DEBUG:
```
Comment (lzx1413) [inline demo/resources/static_img_example/getImage.py]: should be removed

File: main/debug/debug-dataloader.py [hunk 7]
```diff
@@ -0,0 +1,44 @@
+from paths import ROOT_PATH  # isort:skip
```
Comment (lzx1413) [inline main/debug/debug-dataloader.py]: this file should be removed

File: main/debug/debug-datapipeline.py [hunk 8]
```diff
@@ -0,0 +1,52 @@
+from paths import ROOT_PATH  # isort:skip
```
Comment (lzx1413) [inline main/debug/debug-datapipeline.py]: remove

File: main/debug/debug-loss.py [hunk 9]
```diff
@@ -0,0 +1,62 @@
+from paths import ROOT_PATH  # isort:skip
```
Comment (lzx1413) [inline main/debug/debug-loss.py]: remove

File: main/debug/paths.py [hunk 10]
```diff
@@ -0,0 +1,15 @@
+r"""
```
Comment (lzx1413) [inline main/debug/paths.py]: remove

File: requirements.txt [hunk 11]
```diff
@@ -1,7 +1,7 @@
 tqdm
 pylint
 yacs
-pillow
+pillow==
```
Comment (lzx1413) [inline requirements.txt]: withdraw this change

File: demo/main/debug.py (hunk 1 above)
Comment (MARMOTatZJU) [reply demo/main/debug.py]: fixed

File: demo/main/osdet/osdet_demo.py (hunk 2 above)
Comment (MARMOTatZJU) [reply demo/main/osdet/osdet_demo.py]: fixed

File: demo/main/osdet/osdet_demo2.py (hunk 3 above)
Comment (MARMOTatZJU) [reply demo/main/osdet/osdet_demo2.py]: fixed

File: demo/main/osdet/osdet_demo.py (hunk 2 above)
Comment (MARMOTatZJU) [reply demo/main/osdet/osdet_demo.py]: fixed

File: demo/main/static_image/static_image.py (hunk 4 above)
Comment (MARMOTatZJU) [reply demo/main/static_image/static_image.py]: fixed

File: demo/resources/static_img_example/getImage.py (hunk 5 above)
Comment (MARMOTatZJU) [reply demo/resources/static_img_example/getImage.py]: fixed

File: demo/resources/static_img_example/getImage.py (hunk 6 above)
Comment (MARMOTatZJU) [reply demo/resources/static_img_example/getImage.py]: fixed

File: main/debug/debug-dataloader.py (hunk 7 above)
Comment (MARMOTatZJU) [reply main/debug/debug-dataloader.py]: fixed

File: main/debug/debug-datapipeline.py (hunk 8 above)
Comment (MARMOTatZJU) [reply main/debug/debug-datapipeline.py]: fixed

File: main/debug/debug-loss.py (hunk 9 above)
Comment (MARMOTatZJU) [reply main/debug/debug-loss.py]: fixed

File: main/debug/paths.py (hunk 10 above)
Comment (MARMOTatZJU) [reply main/debug/paths.py]: fixed

File: requirements.txt (hunk 11 above)
Comment (MARMOTatZJU) [reply requirements.txt]: fixed

**Comment by MARMOTatZJU** [comment]
All aforementioned requests for modification have been applied. Commits will be updated immediately.

**Comment by MARMOTatZJU** [comment]
Commits have been pushed.
